### PR TITLE
Retry booting redshift if we get a 503

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -87,7 +88,14 @@ def _redshift_database_tool():
...
     session = requests.Session()
-    resp = session.post('https://bigcrunch.herokuapp.com/session/')
+
+    while True:
+        resp = session.post('https://bigcrunch.herokuapp.com/session/')
+        if resp.status_code != 503:
+            break
+        print('waiting for Redshift to boot up')
+        time.sleep(15)
+
```
Comment (jklukas) [inline tests/conftest.py]: Will Travis eventually kill the tests if they run too long? Do we know what that timeout is?

**Comment by jklukas** [comment]
The test failures now are due to `HTTPError: 500 Server Error: Internal Server Error`.

**Comment by graingert** [comment]
@jklukas yeah that's a bug on BigCrunch

File: tests/conftest.py (hunk 1 above)
Comment (graingert) [inline tests/conftest.py]: yeah it's like 15 minutes. But it's free compute time ;)

**Comment by graingert** [comment]
@jklukas we get the 500 because we're trying to connect to redshift as soon as the cluster is online, but don't wait for the DNS to propagate.  I just need to put yet another retry around it.

With this change , we only need to rebuild the travis job once, rather than 2-3 times.

**Pull request merged**